src/rewriters/Rewriter.h

```diff
@@ -46,30 +48,29 @@ class Rewriter {
                 continue;
             }
             // If we are here, we have already processed all children
-            assert(done[Idx(term.getId())] == 0);
             vec<PTRef> newArgs(childrenCount);
             bool needsChange = false;
-            for (unsigned i = 0; i < childrenCount; ++i) {
-                auto it = substitutions.find(term[i]);
-                bool childChanged = it != substitutions.end();
+            for (auto i = 0; i < newArgs.size(); ++i) {
```

Comment:
Why not keep using `childrenCount` here when we already have the size there?

Reply:
Fixed.